File: 1528-Shuffle-String/1528-Shuffle-String.java
class Solution {
    public String restoreString(String s, int[] indices) {
        StringBuilder ans = new StringBuilder("");
        for(int i = 0; i < s.length(); i++){
            ans.append('x');
        }
        System.out.println(ans);
        for(int i = 0; i < s.length(); i++){
            ans.setCharAt(indices[i], s.charAt(i));
        }
        return String.valueOf(ans);
    }
}

File: 1528-Shuffle-String/1528-Shuffle-String.py
class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        d = {}
        for i in range(len(s)):
            d[indices[i]] = s[i]
        ans = ""
        for i in range(len(s)):
            ans += d[i]
        return ans